### PR TITLE
Azure Library Workitem Service

### PR DESCRIPTION
**Pull request by gkillough**

### CODE REVIEW

File: azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/util/AzureFieldsExtractor.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.Optional;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+public class AzureFieldsExtractor {
+    private final Gson gson;
+
+    public AzureFieldsExtractor(Gson gson) {
```
Comment (jamesrichard91) [inline azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/util/AzureFieldsExtractor.java:34]: What is the intended use for this class? It looks like it's only used in the tests.

Comment (jamesrichard91) [reply azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/util/AzureFieldsExtractor.java:34]: Are we not able to deserialize the response into java objects correctly?

Comment (gkillough) [reply azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/util/AzureFieldsExtractor.java:34]: It's a utility to extract the actual objects from `fields` fields in certain Azure responses where the `fields` object of the response is dynamic.

Comment (gkillough) [reply azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/util/AzureFieldsExtractor.java:34]: We cannot deserialize them losslessly.

File: azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/service/workitem/WorkItemResponseFields.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.synopsys.integration.azure.boards.common.util.AzureFieldDefinition;
+
+public final class WorkItemResponseFields {
```
Comment (gkillough) [inline azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/service/workitem/WorkItemResponseFields.java:34]: This is a non-exhaustive list of known fields that are possible on a WorkItem.

**Pull request merged**